### ISSUE TITLE
apks will not install

### ISSUE THREAD

**Issue by SteveALee** [legorovers/MarsRoverWebsite#1]
We tried sideloading the lego-rovers.apk and mars-rover.apk but both give the error 'There is a problem parsing the package'

**Comment by louiseadennis** [comment]
Hmm...  we've managed loading onto our Samsungs without problem (at least I uploaded the apk to http://www.csc.liv.ac.uk/~lad/lego-rovers.apk and then downloaded onto the devices without trouble).

We didn't have trouble installing on a Nexus 7 and had to do that, I think, via Android Studio directly.

**Comment by SteveALee** [comment]
Thanks for quick reply. \0/
We tried downloading raw and the full zip and then copied to tablet.

Will try direct download now we know it exists :)

**Comment by SteveALee** [comment]
Same problem!

Could it be an Android version issue? We have Samsung Galaxy Tab

Currently downloading an android update.

**Comment by louiseadennis** [comment]
Yes there is a minimum Android Version requirement - its stated on one of the website pages.  I had to update all the Samsungs first.

**Comment by SteveALee** [comment]
aaaaahhhh haaaahhhh. Andy said he ignore that. :-P

**Comment by SteveALee** [comment]
so RTFM!

**Comment by SteveALee** [comment]
hehe Andy just found out we're we're on 4.1 and it needs 4.4

**Comment by louiseadennis** [comment]
There's quite a bit of fiddling with WiFi and Bluetooth settings you need to do to get it to work as well, because leJOS tries to connect to the brick over Bluetooth using an IP address.

**Comment by JoeCol** [comment]
Hi everyone

There's two requirements for the apk, Android version 4.4 and a screen
density of 550dp which should be the size of a small tablet.

The screen requirement can be removed from the android manifest file when
compiling

Joe Collenette

**Comment by SteveALee** [comment]
Thanks Joe.
Google get null points for that error message!
Still downloading 600MB Android update

**Comment by SteveALee** [comment]
How can we determine the device screen density? can;t find in settings -> info

**Comment by louiseadennis** [comment]
If everything has gone as planned (!?!) you should have the same Samsung model as we were using so the screen density _ought_ to be OK.

**Comment by louiseadennis** [comment]
You'll also need to switch your linux SD card in the EV3 for one with leJOS on.  I think we have the right image in the website files somewhere, but if not there's a Dropbox I can share with you.

**Comment by SteveALee** [comment]
We have the image up and running. Can't connect from Mac though. Is paired over bluetooth.
Andy used the 0.9 version from laos site.  is that correct?

**Comment by louiseadennis** [comment]
It should be connecting directly to the Robot - no Macs should be involved.

**Comment by SteveALee** [comment]
There is a client that runs on the Mac. We are trying that while waiting for tablet update. Not help as the download just got cancelled

**Comment by louiseadennis** [comment]
Hmm... well the app is expecting a bluetooth connection via a paired bluetooth device which it is using for internet connectivity.  I've no idea what it will do if there's a mac at the other end though - especially as once it has the connection it tries to connect to motors and sensors....

**Comment by SteveALee** [comment]
Sorry I'm not being clear. We are running the app on the Mac. not the tablet. The brick has LaJOS on.
At current rate tablet will update tomorrow. :(

We paired both Mac and Tablet via bluetooth

is LeJOS 0.9 correct?

**Comment by louiseadennis** [comment]
Yes, 0.9, not sure if we used 0.9.0 or 0.9.1 though - I seem to have both downloaded and the actual image file for the EV3 doesn't seem to have version numbers anywhere obvious.

**Comment by SteveALee** [comment]
Great!

**Comment by louiseadennis** [comment]
You could try the code in the LegoDinos repository first maybe?  That definitely works from a Mac (running Mavericks)

**Comment by SteveALee** [comment]
OK, thanks.
